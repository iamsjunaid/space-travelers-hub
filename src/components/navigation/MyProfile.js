import React from 'react';
import { useSelector } from 'react-redux';
import RocketsTable from './RocketsTable';
import DragonsTable from './DragonsTable';
import MissionsTable from './MissionsTable';
import '../../styles/MyProfile.css';

function MyProfile() {
  const { missions } = useSelector((state) => state.missions);
  const reservedMissions = missions.filter((mission) => mission.reserved === true);
  const { rockets } = useSelector((state) => state.rockets);
  const reservedRockets = rockets.filter((rocket) => rocket.reserved === true);
  const dragonsState = useSelector((state) => state.dragons);
  // eslint-disable-next-line no-unneeded-ternary
  const { dragons } = dragonsState ? dragonsState : { dragons: [] };
  const reserveDragon = dragons.filter((dragon) => dragon.reserved === true);

  return (
    <div className="my-profile">
      <div className="rockets-profile">
        <h3>My Rockets</h3>
        {reservedRockets.length > 0 ? (
          <RocketsTable rockets={reservedRockets} className="rockets-table" />
        ) : (
          <div>No Rockets Reserved</div>
        )}
      </div>
      <div className="missions-profile">
        <h3>My Missions</h3>
        {reservedMissions.length > 0 ? (
          <MissionsTable missions={reservedMissions} className="missions-table" />
        ) : (
          <div>No Missions Reserved</div>
        )}
      </div>
      <div className="dragons-profile">
        <h3>My Dragons</h3>
        {reserveDragon.length > 0 ? (
          <DragonsTable dragons={reserveDragon} className="dragons-table" />
        ) : (
          <div>No Dragons Reserved</div>
        )}
      </div>
    </div>
  );
}

export default MyProfile;
