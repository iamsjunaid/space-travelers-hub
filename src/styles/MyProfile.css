.rocketsProfile h3 {
  font-weight: normal;
}
