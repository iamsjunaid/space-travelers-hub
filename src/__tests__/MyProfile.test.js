import React from 'react';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { configureStore } from '@reduxjs/toolkit';
import MyProfile from '../components/navigation/MyProfile';
import rocketsReducer from '../redux/rockets/rocketsSlice';

const store = configureStore({
  reducer: {
    rockets: rocketsReducer,
  },
});

test('renders ProfileRockets correctly', () => {
  const rockets = [
    {
      id: 1,
      rocket_name: 'Falcon 9',
      description: 'A two-stage rocket designed and manufactured by SpaceX.',
      image: 'falcon9.jpg',
      reserved: false,
    },
    {
      id: 2,
      rocket_name: 'Falcon 10',
      description: 'A two-stage rocket designed and manufactured by SpaceX.',
      image: 'falcon10.jpg',
      reserved: true,
    },
  ];

  const dragons = [
    {
      id: 'dragon1',
      name: 'Dragon 1',
      description: 'Dragon is a reusable spacecraft developed by SpaceX, an American private space transportation company based in Hawthorne, California.',
      image: 'https://i.imgur.com/9fWdwNv.jpg',
      reserved: false,
    },
    {
      id: 'dragon2',
      name: 'Dragon 2',
      description: 'Dragon 2 (also Crew Dragon, Dragon V2, or formerly DragonRider) is the second version of the SpaceX Dragon spacecraft, which will be a human-rated vehicle.',
      image: 'https://i.imgur.com/9fWdwNv.jpg',
      reserved: true,
    },
  ];

  const { container } = render(
    <Provider store={store}>
      <MyProfile rockets={rockets} dragons={dragons} />
    </Provider>,
  );

  expect(container.firstChild).toMatchSnapshot();
  expect(screen.getByText('My Rockets')).toBeInTheDocument();
  expect(screen.getByText('My Dragons')).toBeInTheDocument();
});
