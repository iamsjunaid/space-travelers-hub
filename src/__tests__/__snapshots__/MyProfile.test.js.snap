// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ProfileRockets correctly 1`] = `
<div
  class="rocketsProfile"
>
  <h3>
    My Rockets
  </h3>
  <div>
    No Rockets Reserved
  </div>
</div>
`;
